sdk/lib/_http/http_impl.dart

```diff
@@ -2285,6 +2288,11 @@ class _HttpClient implements HttpClient {
     }
 
     bool isSecure = (uri.scheme == "https");
+    bool isMobileClient = Platform.isAndroid || Platform.isIOS;
```

Comment:
If you remove the restriction that the flag only works on mobile platforms, then I suspect it will be easier to write tests. A regular Dart program could just install an HttpOverrides in a zone with the flag flipped, and check that the right exception is thrown.

Reply:
There are two reason for including a mobile client distinction:
- We intend to flip the default in google3 but only for mobile. The default should be `allow` for non-mobile applications even in google3. Perhaps, this can be solved by having defaults per Platform.
- It reflects the principle of the change more accurately. On iOS and Android, we should not be asking the apps to install an Override to get this behavior. Both OSes ban cleartext HTTP by default for their native apps.

I introduced platform specific defaults to capture both points and removed the `isMobileClient` check.